Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnSearch.hs

```diff
@@ -215,7 +220,61 @@ mkQuoteEntities start end provider estInfo = do
                 ]
             }
 
-mkQuoteEntitiesSpecialZone :: OS.StartInfo -> OS.StopInfo -> DM.Merchant -> DSearch.SpecialZoneQuoteInfo -> QuoteEntities
+mkQuoteEntitiesRental :: OS.StartInfo -> DM.Merchant -> DSearch.RentalQuoteInfo -> QuoteEntities
+mkQuoteEntitiesRental start provider it = do
+  let variant = Common.castVariant it.vehicleVariant
+      baseFare = OS.DecimalValue $ toRational it.baseFare
+      baseDistance = it.baseDistance
+      baseDuration = it.baseDuration
+      fulfillment =
+        OS.FulfillmentInfo
+          { start,
+            end = Nothing,
+            id = it.quoteId.getId,
+            _type = OS.RENTAL,
+            vehicle = OS.Vehicle {category = variant}
+          }
+      item =
+        OS.Item
+          { id = Common.mkItemId provider.shortId.getShortId it.vehicleVariant,
+            fulfillment_id = fulfillment.id,
+            price =
+              OS.ItemPrice
+                { currency = currency',
+                  value = baseFare,
+                  offered_value = baseFare,
+                  minimum_value = baseFare,
+                  maximum_value = baseFare
+                },
+            tags = Just $ OS.TG [mkRentalTag baseDistance baseDuration]
+          }
+  QuoteEntities
+    { fulfillment,
+      item
+    }
+  where
+    mkRentalTag baseDistance baseDuration =
+      OS.TagGroup
+        { display = False,
+          code = "general_info",
+          name = "General Information",
+          list =
+            [ OS.Tag
+                { display = Just True,
+                  code = Just "rental_base_duration",
+                  name = Just "Base Duration",
+                  value = Just $ show baseDuration.getHours
+                },
+              OS.Tag
+                { display = Just True,
+                  code = Just "rental_base_distance",
+                  name = Just "Base Distance",
+                  value = Just $ show baseDistance.getKilometers
+                }
```

Comment:
Need to pass other params too, for example: `nightShiftInfo`, `perKmCharge`, `perMinCharge` etc.

Reply:
@0utkarsh sure for **nightShiftInfo** 
but for **perKmCharge, perMinCharge** shouldn't this params be taken from farePolicyDetails of FarePolicy in domain level just how rest of the params are taken